contracts/libraries/SemverLibrary.sol

```diff
@@ -0,0 +1,143 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.9;
+
+library SemverLibrary {
+    uint8 internal constant ASCII_ZERO = 48;
+    uint8 internal constant BIT_OFFSET = 85;
+    uint256 internal constant MAX_LENGTH = 0x1f;
+
+    function isNumeric(bytes1 num) internal pure returns (bool) {
+        return num >= "0" && num <= "9";
+    }
+
+    function shrinkToFit(bytes memory array) internal pure returns (bytes memory result) {
+        uint256 i;
+        while (i != array.length && array[i] != 0) {
+            ++i;
+        }
+        result = new bytes(i);
+        for (uint256 j; j != i; ++j) {
+            result[j] = array[j];
+        }
+    }
+
+    function numberify(bytes memory _num) internal pure returns (uint256 result) {
+        for (uint256 i; i != _num.length; ++i) {
+            result *= 10;
+            result += uint256(uint8(_num[i])) - ASCII_ZERO;
+        }
+    }
+
+    function stringify(uint256 num) internal pure returns (bytes memory result) {
```

Comment:
There's Strings.toString in openzeppelin utils

Reply:
strings are immutable 
I don't want to do an extra cast